client/src/Components/local/atoms/core-mapping/index.tsx

```diff
@@ -107,11 +109,59 @@ const CoreMapping: React.FC<PropTypes> = ({ messages, channel, playerForce, play
         details, 
         featureCollection,
         messageType: CUSTOM_MESSAGE,
-        _id: timestamp
+        _id: lastMessage ? lastMessage._id : timestamp
+      }
+
+      if (lastMessage) {
+        const diffKeyArray: string[] = getObjDifferences(lastMessage, newMessage, [], '')
+        const diffKeyArray2: string[] = getObjDifferences(newMessage, lastMessage, [], '')
+        // build message deltas from diff keys array
+        const delta = {};
+        [...diffKeyArray, ...diffKeyArray2].forEach(key => {
+          if (hasIn(newMessage, key)) {
+            set(delta, key, get(newMessage, key))
+          }
+        })
+        const deltaMessage: MappingMessageDelta = {
+          _id: new Date().toISOString(),
+          messageType: MAPPING_MESSAGE_DELTA,
+          details: {
+            channel: lastMessage.details.channel,
+            from: {
+              force: '',
+              forceColor: '',
+              roleId: '',
+              roleName: '',
+              iconURL: ''
+            },
+            messageType: '',
+            timestamp: '',
+            turnNumber: 0
+          },
+          since: newMessage._id,
+          delta
+        }
+        postBack(deltaMessage)
       }
-      postBack(newMessage as MappingMessage)
     }
   }
+
```

Comment:
@lebaphi - could we please generate the diff using an JSON-Patch library?  Then we can be confident of how the diff is written/implemented. This also means apps other than Serge may be able to handle/generate them.
The possible libraries are here:
https://jsonpatch.com/#javascript

Reply:
I just added the jiff lib to generate the diff but it seems the lib throws an error when applying the patch, I'm still investigating it. @IanMayo

Reply:
I decided to use this lib https://www.npmjs.com/package/fast-json-patch, it seems better in terms of performance and it solves the error of jiff lib. The only small problem I have is that after the c/u/d feature and reloading the page, all the features are gone. I'm still working on it but I welcome you to check this out. @IanMayo

Reply:
I think I've got it. Our current `delta` handling is in `handleNewMessage`.  That method doesn't get called when we load an existing wargame.
So, I think the `delta` handling has to happen in `CoreMapping`.  Then it will run on initial load aswell as on new messages.